windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -996,17 +996,28 @@ function InstallingState.DoInstallPackage(PackageFileName: String): Boolean;
 var
   FPackage: IKeymanPackageFile2;
 begin
-  Result := True;
-  FPackage := kmcom.Packages.GetPackageFromFile(PackageFileName)
-    as IKeymanPackageFile2;
-  FPackage.Install2(True);
-  // Force overwrites existing package and leaves most settings for it intact
-  FPackage := nil;
-
-  kmcom.Refresh;
-  kmcom.Apply;
+  Result := False;
+  try
+    FPackage := kmcom.Packages.GetPackageFromFile(PackageFileName)
+      as IKeymanPackageFile2;
+      // Force overwrites existing package and leaves most settings for it intact
+    FPackage.Install2(True);
+    Result := True;
+  except
+    on E: Exception do
+      begin
+        TKeymanSentryClient.ReportHandledException(E,
+          'Failed to install keyboard package');
+      end;
+  end;
 
-  System.SysUtils.DeleteFile(PackageFileName);
+  FPackage := nil;
+  if Result then
```

Comment:
Marc I wasn't sure if this is the best way to do the exception can I just put ln:1015 to 1019 inside the try block?

Reply:
Hmm. What is the exception? From DeleteFile? Or elsewhere?

It's best to have very narrowly-scoped try/except blocks.